### ISSUE TITLE
Docker questions and lsb_release: command not found

### ISSUE THREAD

**Issue by bufke** [joshjdevl/libsodium-jni#72]
If I run `docker-compose -f docker-compose-build.yml build` I get an error.

```
lsb_release -a
./dependencies-linux.sh: line 7: lsb_release: command not found
```

It can be avoided by installing lsb-release in the Dockerfile before running the dependencies step. Should this be done? Also is it even necessary to build the Dockerfile or should the existing joshjdevl/libsodium-jni image allow me to build the source? I'm happy to contribute to documentation on getting started with building via docker.

I was also wondering what you think of separating the Dockerfile steps (RUN statements) out more. The ./build-linux.sh step is huge. If we run it in a few different steps it would be easier to use cache layers so that not everything must be done at once. This would make it easier to deal with problems happening at the end. I'd propose at least putting dependencies-linux.sh as it's own step.

**Comment by joshjdevl** [comment]
> It can be avoided by installing lsb-release in the Dockerfile before running the dependencies step. Should this be done?

Thank you for brining this to my attention. It appears that the docker build broke. I have pushed changes to make sure 'lsb_release' is installed.

> Also is it even necessary to build the Dockerfile or should the existing joshjdevl/libsodium-jni image allow me to build the source?

It should be possible to pull directly from dockerhub. No need to rebuild the docker container.

> I'm happy to contribute to documentation on getting started with building via docker.

That would be very helpful to help tighten these edge cases!

> I was also wondering what you think of separating the Dockerfile steps (RUN statements) out more. The ./build-linux.sh step is huge. If we run it in a few different steps it would be easier to use cache layers so that not everything must be done at once. This would make it easier to deal with problems happening at the end. I'd propose at least putting dependencies-linux.sh as it's own step.

The reason for one file is to ensure consistency across environments. That is one file to install dependencies and one to build. If anything needs to change, update, or be added the callers of these scripts don't need to change. For example, Dockerfiles and [Travis](https://github.com/joshjdevl/libsodium-jni/blob/master/.travis.yml).

The reason I did not separate each command of the script into Docker RUN commands is because would like to make sure that the build process is the same steps regardless of it's Docker, Virtual Machine, or bare bones OS. The Dockerfile RUN commands are specific only to Docker and not callable from plain Ubuntu, for example. The only other alternative I'm aware of is Vagrant which is too much at this point.

Is there anything new in Docker land which attempts to solve this consistent build problem?
